server/server.go

```diff
@@ -2981,3 +3025,14 @@ func annotateSpan(ctx context.Context, decisionID string) {
 	trace.SpanFromContext(ctx).
 		SetAttributes(attribute.String(otelDecisionIDAttr, decisionID))
 }
+
+func readPlainBody(r *http.Request) (io.ReadCloser, error) {
+	if strings.Contains(r.Header.Get("Content-Encoding"), "gzip") {
+		plainBody, err := gzip.NewReader(r.Body)
+		if err != nil {
+			defer plainBody.Close()
```

Comment:
Uhm... Shouldn't we close the `plainBody` reader if there was _no_ error? 🤔 Like
```go
plainBody, err := gzip.NewReader(r.Body)
if err != nil {
	return nil, err
}
defer plainBody.Close()
```
but also, is this really what we want? We can't have the caller _read the io.ReadCloser_ if we close if after returning it.

Reply:
you are right, I've removed the close & the faulty `if` condition. the err is managed by the caller

Reply:
Now the question remains  -- who's going to Close() it? I think that's also going to be the caller, but I wasn't sure if that was handle in the code right now.

Reply:
@srenatus I need your input here please.

Reply:
Sounds good to me, sorry for letting you wait

Reply:
@srenatus thank you, no problem, it is ready for another review